server/scsynth/SC_Graph.cpp

```diff
@@ -431,15 +431,28 @@ void Graph_Ctor(World *inWorld, GraphDef *inGraphDef, Graph *graph, sc_msg_iter
 	inGraphDef->mRefCount ++ ;
 }
 
+// Calls the unit's Ctor func.
+inline void InitializeUnit(Unit* unit)
+{
+	try {
+		(*unit->mUnitDef->mUnitCtorFunc)(unit);
```

Comment:
a custom exception will have the same problem as `std::exception`. one could add an API function to the interface table to throw an exception. this would prevent lookup issues, but won't help with issues arising from going through a plain c interface